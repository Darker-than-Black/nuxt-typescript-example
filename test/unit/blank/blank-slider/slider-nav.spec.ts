import VueTestInstance from '~/test/factory/vue-instance'
import SliderNav from '~/components/blank/items/blank-slider/slider-nav.vue'

describe('slider-nav.vue', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(SliderNav, {
      options: {
        propsData: {
          sliderSize: 10
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit input() return current slide', () => {
    expect(wrapper.vm.input()).toBe(wrapper.vm.currentSlide)
  })

  it('test nextSlide()', () => {
    wrapper.vm.input = jest.fn()
    wrapper.vm.nextSlide()
    expect(wrapper.vm.input).toHaveBeenCalled()
    expect(wrapper.vm.currentSlide).toBe(2)
  })

  it('test prevSlide()', () => {
    wrapper.vm.input = jest.fn()
    wrapper.vm.nextSlide()
    wrapper.vm.prevSlide()
    expect(wrapper.vm.input).toHaveBeenCalled()
    expect(wrapper.vm.currentSlide).toBe(1)
  })
})
