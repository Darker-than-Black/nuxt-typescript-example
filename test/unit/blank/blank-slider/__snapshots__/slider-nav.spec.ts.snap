// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slider-nav.vue should match the snapshot 1`] = `
<div class="slider-nav"><button disabled="disabled" class="btn btn-flex btn-transparent small mr-2"><img src="/icons/Arrow-right.svg" alt="next" class="img img__rotate left"></button> <span class="mr-2 slider-nav__text">
    1 / 10
  </span> <button class="btn btn-flex btn-transparent small"><img src="/icons/Arrow-right.svg" alt="next" class="img"></button></div>
`;
